tensorflow/compiler/mlir/hlo/lib/Dialect/lhlo/transforms/lhlo_legalize_to_affine.cc

```diff
@@ -462,6 +462,107 @@ class GatherOpConverter : public OpRewritePattern<GatherOp> {
   }
 };
 
+/// Converts PadOp to Affine nest form.
+/// Pseudocode:
+///   1. Fill `output` tensor with `padding_value`.
+///   2. Compute AffineMap for store into `output`.
+///      out_idx = edge_padding_low +
+///                operand_idx * (1 + interior_padding)
+///   3. Create nested loop from `operand` shape.
+///      3.1 load from `operand`.
+///      3.2 store into `output`.
+/// Limitation for now:
+///   interior_padding == 0 && edge_padding_* >= 0
+struct PadOpConverter : public OpRewritePattern<PadOp> {
+  using OpRewritePattern<PadOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(PadOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value operand = op.operand();
+    Value padding_value = op.padding_value();
+    Value output = op.output();
+
+    auto operand_type = operand.getType().dyn_cast<ShapedType>();
+    auto output_type = output.getType().dyn_cast<ShapedType>();
+    // We allow lowering for only ranked input/output.
+    if (!(operand_type && output_type && operand_type.hasRank() &&
+          output_type.hasRank()))
+      return failure();
+    unsigned rank = operand_type.getRank();
+    assert(output_type.getRank() == rank);
+
+    auto edge_padding_low_it = op.edge_padding_low().getValues<int64_t>();
+    auto edge_padding_high_it = op.edge_padding_high().getValues<int64_t>();
+    auto interior_padding_it = op.interior_padding().getValues<int64_t>();
+    SmallVector<int64_t> edge_padding_low(edge_padding_low_it.begin(),
+                                          edge_padding_low_it.end());
+    SmallVector<int64_t> edge_padding_high(edge_padding_high_it.begin(),
+                                           edge_padding_high_it.end());
+    SmallVector<int64_t> interior_padding(interior_padding_it.begin(),
+                                          interior_padding_it.end());
+
+    assert(edge_padding_low.size() == rank &&
+           edge_padding_high.size() == rank && interior_padding.size() == rank);
```

Comment:
same assert problem as above.

Reply:
Instead of asserting on it I've included a check which bails out on lowering instead by returning `failure()` if the check fails.